templates/rust/src/client.rs.twig

```
@@ -0,0 +1,254 @@
+                                form = form.text(k, data.to_string());
+                            }
+                            ParamType::String(data) => form = form.text(k, data),
+                            ParamType::FilePath(data) => form = form.file(k, data).unwrap(),
+                            ParamType::Number(data) => form = form.text(k, data.to_string()),
+                            // This shouldn't be possible due to the flatten function, so we won't handle this for now
+                            ParamType::Array(_data) => {
+                                //todo: Feed this back into a flatten function if needed
+                            },
+                            ParamType::Object(_data) => {
+                              // Same for this
+                            }
+                        }
+                    }
+                    request = request.multipart(form);
+                }
+
+                if content_type.starts_with("application/json") && method != "GET" {
+                    request = request.json(&data);
+                }
+
+                if method == "GET" {
+                    request = request.query(&queryize_data(flatten(FlattenType::Normal(data), None)));
+                }
+            }
+            None => {}
+        }
+
+        request = request.headers(request_headers);
+
+        match request.send() {
+            Ok(data) => {
+                if data.status().is_success() {
+                    Ok(data)    
+                } else {
+                    // Format error
+                    Err(match data.json() {
+                        Ok(data) => data,
+                        Err(err) => {
+                            {{spec.title | caseUcfirst}}Exception::new(format!("A error occoured. ERR: {}, This could either be a connection error or an internal Appwrite error. Please check your Appwrite instance logs. ", err), 0, "".to_string())
```

Here the error message should be the one received from the API `data` i presume.

---

Yes .json() already turns it into a AppwriteException so we can just return it into the Err() block

---

@PineappleIOnic yes instead of custom message, it should return the error message in the json Data